From 1234e5240bc690558b3e19de0830e407b76747e9 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Fri, 19 Sep 2014 18:38:08 +0800
Subject: [PATCH] ENGR00332319 lcdif: imx6sl: disable the pixel rate before
 call clk_set_rate

After the following commit is pushed, the lcdif framebuffer driver need
the adjustment.
93a9e3d0b88203cb523dd92e85590683d6a85fdf ENGR00318063-6:
ARM: imx6: add CLK_SET_RATE_GATE flag for PLL clocks

CLK_SET_RATE_GATE flag means "must be gated across rate change".

PLL5 video is the parent clock of the pixel clock, and only used by it.
This patch gates the clock before call clk_set_rate() to meet the requirement.

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 drivers/video/mxsfb.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index 59ea7e4..f7fae186 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -494,6 +494,22 @@ static void mxsfb_enable_controller(struct fb_info *fb_info)
 	clk_enable_disp_axi(host);
 
 	clk_set_rate(host->clk_pix, PICOS2KHZ(fb_info->var.pixclock) * 1000U);
+	ret =
+	    clk_set_rate(host->clk_pix,
+			 PICOS2KHZ(fb_info->var.pixclock) * 1000U);
+	if (ret) {
+		dev_err(&host->pdev->dev,
+			"lcd pixel rate set failed: %d\n", ret);
+
+		if (host->reg_lcd) {
+			ret = regulator_disable(host->reg_lcd);
+			if (ret)
+				dev_err(&host->pdev->dev,
+					"lcd regulator disable failed: %d\n",
+					ret);
+		}
+		return;
+	}
 	clk_prepare_enable(host->clk_pix);
 
 	/* Clean soft reset and clock gate bit if it was enabled  */
-- 
1.8.0

